src/Compilers/VisualBasic/Portable/Symbols/Source/SourceMemberMethodSymbol.vb

```diff
@@ -743,6 +743,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     diagBag)
             End If
 
+            Select Case ContainingType.TypeKind
+                Case TypeKind.Interface, TypeKind.Structure, TypeKind.Enum, TypeKind.Delegate
+                    ' Handles clause is invalid in this context. 
+                    Return Nothing
+
+                Case TypeKind.Class, TypeKind.Module
+                    ' Valid context
+
+                Case Else
+                    Debug.Assert(False)
```

Comment:
Yes, there is a very good chance that it would just work. For example, it would work fine for Structures. It wouldn't for an Interface because an Interface doesn't have a constructor.